tests/system.rs

```diff
@@ -2908,6 +2908,61 @@ fn ignores_impl_macro() {
 }
 
 #[test]
+fn try_operator() {
+    let _lock = sync!();
+
+    let src = "
+        pub struct Foo(u16);
+
+        #[derive(Debug, Clone, PartialEq, Eq)]
+        pub struct OddError;
+
+        fn be_even(val: Foo) -> Result<Foo, OddError> {
+            if val.0 % 2 == 1 {
+                Err(OddError)
+            } else {
+                Ok(val)
+            }
+        }
+
+        pub fn half(val: Foo) -> Result<Foo, OddError> {
+            Ok(Foo(be_even(val)?.~0 / 2))
+        }
+    ";
+
+    let got = get_definition(src, None);
+    assert_eq!("0", got.matchstr);
```

Comment:
What do you think about adding a completions test with a non-tuple struct? I imagine like it works, but these tests don't cover that case.